### PR TITLE
feat(uat): implement scenario outline GGMQ-1-T27

### PR DESCRIPTION
**Pull request by bgklika**
**Issue #, if available:**
https://klika-tech.atlassian.net/browse/GGMQ-208
Implement GGMQ scenario based on GGAD-1-T27

**Description of changes:**
- Add GGMQ-1-T27 scenario outline
- Move list of addresses of brokers from MqttControlSteps to MqttBrokers
- Add BrokerCertificateSteps with implementation of certificate getting and comparision
- In addition: update readme with description of tags

**Why is this change necessary:**
Implement replacement of GGAD-1-T27 scenario outline

**How was this change tested:**
Scenarios have been run on Linux for 4 clients (later reduced to 1) and Windows (one client) for both MQTT v3 and v5.

**Test results:**
Linux:
```
[INFO ] 2023-07-25 22:46:59.173 [main] StepTrackingReporting - Passed: 'GGMQ-1-T27-v3-sdk-java: As a customer, my Greengrass-issued certificate does not rotate when mqtt reconnects'
[INFO ] 2023-07-25 22:46:59.173 [main] StepTrackingReporting - Passed: 'GGMQ-1-T27-v5-sdk-java: As a customer, my Greengrass-issued certificate does not rotate when mqtt reconnects'
```

Windows:
```
[INFO ] 2023-07-25 23:22:47.231 [main] StepTrackingReporting - Passed: 'GGMQ-1-T27-v3-sdk-java: As a customer, my Greengrass-issued certificate does not rotate when mqtt reconnects'
[INFO ] 2023-07-25 23:22:47.231 [main] StepTrackingReporting - Passed: 'GGMQ-1-T27-v5-sdk-java: As a customer, my Greengrass-issued certificate does not rotate when mqtt reconnects'
```

**Any additional information or context required to review the change:**

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: uat/testing-features/src/main/java/com/aws/greengrass/steps/BrokerCertificateSteps.java [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+                certificates.put(certName, cert);
+                log.info("Saved broker's '{}' certificate '{}'", brokerId, cert);
+                return;
+            } catch (IllegalStateException | ExecutionException | TimeoutException ex) {
+                lastException = ex;
+            }
+        }
+        if (lastException == null) {
+            throw new IllegalStateException("No addresses to get certificate");
+        } else if (lastException instanceof RuntimeException) {
```
Comment (auarbekov) [inline uat/testing-features/src/main/java/com/aws/greengrass/steps/BrokerCertificateSteps.java]: Judging by the try-catch block, the RuntimeException will never be caught, so this part of the code will never be used.

File: uat/testing-features/src/main/java/com/aws/greengrass/steps/BrokerCertificateSteps.java [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+
+                X509Certificate cert = serverCerts[0];
+                certificates.put(certName, cert);
+                log.info("Saved broker's '{}' certificate '{}'", brokerId, cert);
+                return;
+            } catch (IllegalStateException | ExecutionException | TimeoutException ex) {
+                lastException = ex;
+            }
+        }
+        if (lastException == null) {
```
Comment (auarbekov) [inline uat/testing-features/src/main/java/com/aws/greengrass/steps/BrokerCertificateSteps.java:141]: could you please add a comment why the null exception is not logically correct.

**Review by eschastlivtsev — DISMISSED** [review]

File: uat/testing-features/src/main/java/com/aws/greengrass/steps/BrokerCertificateSteps.java (hunk 1 above)
Comment (bgklika) [reply uat/testing-features/src/main/java/com/aws/greengrass/steps/BrokerCertificateSteps.java]: It comes from replacement RuntimeException to IllegalStateException but that place was skipped.
Fixed in e1cede5de9a5ae4e77b39a8fb35f97e0d401406f

File: uat/testing-features/src/main/java/com/aws/greengrass/steps/BrokerCertificateSteps.java (hunk 2 above)
Comment (bgklika) [reply uat/testing-features/src/main/java/com/aws/greengrass/steps/BrokerCertificateSteps.java:141]: Added in e1cede5de9a5ae4e77b39a8fb35f97e0d401406f

**Review by auarbekov — DISMISSED** [review]

**Comment by bgklika** [comment]
@mzyl13  Please review README changes joined from your PR 123

**Review by eschastlivtsev — DISMISSED** [review]

**Review by eschastlivtsev — DISMISSED** [review]

**Pull request closed without merging**